pkg/rtc/mediatrackreceiver.go

```diff
@@ -568,9 +568,15 @@ func (t *MediaTrackReceiver) TrackInfo(generateLayer bool) *livekit.TrackInfo {
 					ci.Layers = append(ci.Layers, proto.Clone(layer).(*livekit.VideoLayer))
 
 					// if origin layer has ssrc, don't override it
-					if layerIdx < len(originLayers) && originLayers[layerIdx].Ssrc != 0 {
-						ci.Layers[layerIdx].Ssrc = originLayers[layerIdx].Ssrc
-					} else if int(layer.Quality) < len(receiver.layerSSRCs) {
+					ssrcFound := false
+					for _, l := range originLayers {
+						if l.Quality == ci.Layers[layerIdx].Quality {
```

Comment:
The older code also had a check for `l.Ssrc != 0`. My diff looks like
```
+                                       originHasSsrc := false
+                                       for _, originLayer := range originLayers {
+                                               if originLayer.Quality != layer.Quality {
+                                                       continue
+                                               }
+
+                                               if originLayer.Ssrc != 0 {
+                                                       // if origin layer has ssrc, don't override it
+                                                       ci.Layers[layerIdx].Ssrc = originLayer.Ssrc
+                                                       originHasSsrc = true
+                                                       break
+                                               }
+                                       }
+
+                                       if !originHasSsrc && int(layer.Quality) < len(receiver.layerSSRCs) {
                                                ci.Layers[layerIdx].Ssrc = receiver.layerSSRCs[layer.Quality]
                                        }
```

Reply:
Actually, the break should be outside the `if` which sets the `ssrc`. As long as quality match is found, whether setting `ssrc` or not, it should break.